test/parallel/test-net-write-after-close.js

```diff
@@ -21,24 +21,27 @@
 
 'use strict';
 const common = require('../common');
+
 const net = require('net');
 
+let serverSocket;
+
 const server = net.createServer(common.mustCall(function(socket) {
+  serverSocket = socket;
+
   socket.resume();
 
   socket.on('error', common.mustCall(function(error) {
     console.error('got error, closing server', error);
     server.close();
   }));
-
-  setTimeout(common.mustCall(function() {
-    console.error('about to try to write');
-    socket.write('test', common.mustCall());
-  }), 250);
 }));
 
 server.listen(0, function() {
   const client = net.connect(this.address().port, function() {
+    client.on('end', () => {
+      serverSocket.write('test', common.mustCall());
+    });
     client.end();
```

Comment:
can you add a comment on top of this? Something like:

```js
// cliend.end() will close both the readable and writable side
// of the duplex, because allowHalfOpen is defaulted to false.
// Then 'end' will be emitted when it receives a FIN packet from
// the other side.
```